src/menu/main_menu.rs

```diff
@@ -0,0 +1,162 @@
+use bevy::prelude::*;
+
+use crate::menu::{ClickAction, MenuMaterials, StateCleanup};
+use crate::AppState;
+
+pub fn setup(
+    commands: &mut Commands,
+    asset_server: Res<AssetServer>,
+    materials: Res<MenuMaterials>,
+) {
+    let button_style = Style {
+        size: Size::new(Val::Auto, Val::Auto),
+        margin: Rect::all(Val::Px(5.0)),
+        padding: Rect {
+            left: Val::Px(12.0),
+            right: Val::Px(12.0),
+            top: Val::Px(10.0),
+            bottom: Val::Px(10.0),
+        },
+        justify_content: JustifyContent::Center,
+        align_items: AlignItems::Center,
+        ..Default::default()
+    };
+
+    let button_text_style = TextStyle {
+        font: asset_server.load("fonts/sansation/sansation_light.ttf"),
+        font_size: 18.0,
+        color: Color::WHITE,
+    };
+
+    commands
+        .spawn(UiCameraBundle::default())
+        // FIXME re-enable this when https://github.com/bevyengine/bevy/issues/1452 is addressed so the camera despawns again
+        //.with(StateCleanup)
+
+        // Container
+        .spawn(NodeBundle {
+            style: Style {
+                size: Size::new(Val::Percent(100.0), Val::Percent(100.0)),
+                position_type: PositionType::Absolute,
+                flex_direction: FlexDirection::ColumnReverse,
+                justify_content: JustifyContent::Center,
+                align_items: AlignItems::Center,
+                ..Default::default()
+            },
+            material: materials.transparent.clone(),
+            ..Default::default()
+        })
+        .with(StateCleanup)
+        .with_children(|root| {
+            root
+                // Game title
+                .spawn(TextBundle {
+                    text: Text::with_section(
+                        "SOTORA",
+                        TextStyle {
+                            font: asset_server.load("fonts/sansation/sansation_regular.ttf"),
+                            font_size: 25.0,
+                            color: Color::rgb(0.9, 0.9, 0.95),
+                        },
+                        Default::default(),
+                    ),
+                    ..Default::default()
+                })
+                .spawn(TextBundle {
+                    text: Text::with_section(
+                        "a bevy community game",
+                        TextStyle {
+                            font: asset_server.load("fonts/sansation/sansation_light.ttf"),
```

Comment:
It seems brittle to just hardcode paths to assets all over the place like that.

I think you should extend `MenuMaterials` to become a larger `UiConfig` (or something along those lines) to hold all the static things that need to be referenced when building the UIs: styles, font, assets, materials, etc. all together. Load/configure them all at once in a single place at setup time.

It would help the code be less brittle, as there would be a single place for all the UI building stuff. It would also make it more convenient to add more UIs to the game, as you can just reference anything you need from one place.

I think one monolithic resource for this stuff is fine, as this is all stuff that is to be loaded/configured once at startup and then only accessed read-only. We wouldn't win anything by making it more granular.

Reply:
I followed Bevy's button UI example where it only uses a resource for the materials so I just implemented it the same way, but it makes total sense to load all assets centrally and make them available trough in a resource. Fixing this right now! 😄